README.md

```diff
@@ -20,16 +22,97 @@ Through a [JSON document](https://docs.camunda.io/docs/next/components/modeler/d
 * Documentation references
 * Versioning + deprecation information
 
+### [Property Configuration](./docs/DEFINITION.md#properties)
+
+* Type of control
+* Technical binding
+* Labels, description
+* Validation
+* Grouping
+* Conditional activation criteria
+
+
+## Editor Integration
+
+Through a [set of utilities](#related-utilities) element templates get deeply integrated into our editor tooling.
+
+### Create/Append Anything
+
+> Provided [through an extension](https://github.com/bpmn-io/bpmn-js-connectors-extension).
+
+* Palette and context pads have a `...` entry to create / append any element
+* Building blocks can be discovered via search
+* Replace menu is enhanced to work using the same pattern
+
+### Configuration
+
+> Provided [through the properties panel](https://github.com/bpmn-io/bpmn-js-properties-panel).
+
+* Properties panel displays [configuration properties](./docs/PROPERTIES_EDITING.md#readme)
+* Validates inputs according to defined validation rules
+* Shows basic template meta-data
+* Allows [applying, unlinking and removing](./docs/APPLICATION.md#readme) of a template
+
+### Upgrading
+
+> Provided [through the properties panel](https://github.com/bpmn-io/bpmn-js-properties-panel) as well as [an extension](https://github.com/bpmn-io/bpmn-js-connectors-extension).
+
+Templates come with two built-in ways of migration: _Upgrading_ to a newer (compatible) version or _deprecation_ (with the ability to display a custom message).
+
+* Upgrading works alongside the built-in [versioning feature](./docs/VERSIONING.md#readme)
+* Migration happens using the built-in [replace feature](./docs/APPLICATION.md#replace)
+* Upgrade and replace will [preserve defined properties](./docs/APPLICATION.md#upgrade-behavior) on a best effort basis (only keeping what is valid in the new template)
+
+
+## Basic Setup
+
+The basic setup requires a BPMN modeler as well as properties panel and element template provider modules:
+
+```javascript
+import BpmnModeler from 'bpmn-js/lib/Modeler';
+
+import {
+  BpmnPropertiesPanelModule,
+  BpmnPropertiesProviderModule,
+  ZeebePropertiesProviderModule,
```

Comment:
```suggestion
```

This is unnecessary as the templates provider already depends on Zeebe provider: https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/src/provider/cloud-element-templates/index.js#L18

We could also consider making Zeebe provider depend on the general BPMN one.

Reply:
Let's leave this here still to show the bits required.